### DISCUSSION TITLE
Ship the console as a standalone application

### DISCUSSION THREAD

**Discussion by abellion** [logto-io/logto#2241] [Ideas]
As far as I know, `logto-io/logto` ships 3 components: an API, a UI for sign-in / sign-up, and a console. The first two components are mandatory in order for the end users to get authenticated, as opposed to the console that is  only made for administrative purposes.

I know that the console is password protected, but in addition to that security, being able to deploy it in a different environment than the API and UI (e.g. behind a specific firewall) would be very welcomed.

We would therefore need the console to be shipped as a standalone application. On top of the security benefit, developing the console outside of the main Logto repository would improve separation of concerns and probably promote collaboration, similar to what was done for `logto-io/connectors`.

What do you think?

**Comment by simeng-li** [comment]
Hey, @abellion thanks for asking.  We actually using monorepo to manage all Logto's core components, like core, UI, and console as you mentioned. It gives us the flexibility to develop and manage those packages individually and a strictly tied-up foundation at the same time.  

Even though they are three different products but deeply connected both product-wise and technical-wise. They all shipped as a bundled solution. We intend to keep them sharing the same foundation works, like interfaces, types, and other development configs.  

It makes the product more robust and manageable.  As a matter of effect, in some scope, it helps our team to collaborate on different packages individually without being afraid of any breaking changes. 

The topic is open for discussion anyway, feel free to send us any specific development issues or use cases you may encounter with.

**Reply by gao-sun** [reply]
I think it's done by [v1.0.0-rc.2](https://github.com/logto-io/logto/releases/tag/v1.0.0-rc.2). Now you can totally disable admin console.
